### ISSUE TITLE
Closed Websocket Server causes SegFault when it gets GC'd

### ISSUE THREAD

**Issue by simadude** [MCJack123/craftos2#376]
**Describe the bug**
SegFault when closed Websocket Server gets garbage collected.

**To Reproduce**
1. Create a Websocket Server using `http.websocketServer()`
2. Close the Websocket Server using `.close()` method.
3. Set the variable to nil.
4. Call Garbage Collector with `collectgarbage("collect")`, or wait until the variable gets GC'd.
5. Segfault.

**Expected behavior**
I guess there should be no SegFault when the closed Websocket Server gets GC'd.

**Screenshots**
```
OS: Linux
Address is 0x80 from 0x4affec
Last C function: websocket_server_free
[bt]: (1) craftos() [0x4affec]
[bt]: (2) craftos() [0x4affec]
[bt]: (3) /nix/store/5ms06w05g7l9y2rakkf5x23p177qhz67-craftos-pc-2.8.3/lib/liblua.so(+0x10149) [0x7f27efc9e149]
[bt]: (4) /nix/store/5ms06w05g7l9y2rakkf5x23p177qhz67-craftos-pc-2.8.3/lib/liblua.so(+0x10626) [0x7f27efc9e626]
[bt]: (5) /nix/store/5ms06w05g7l9y2rakkf5x23p177qhz67-craftos-pc-2.8.3/lib/liblua.so(+0xf67c) [0x7f27efc9d67c]
[bt]: (6) /nix/store/5ms06w05g7l9y2rakkf5x23p177qhz67-craftos-pc-2.8.3/lib/liblua.so(+0x109ff) [0x7f27efc9e9ff]
[bt]: (7) /nix/store/5ms06w05g7l9y2rakkf5x23p177qhz67-craftos-pc-2.8.3/lib/liblua.so(+0x11e4c) [0x7f27efc9fe4c]
[bt]: (8) /nix/store/5ms06w05g7l9y2rakkf5x23p177qhz67-craftos-pc-2.8.3/lib/liblua.so(+0x13901) [0x7f27efca1901]
[bt]: (9) /nix/store/5ms06w05g7l9y2rakkf5x23p177qhz67-craftos-pc-2.8.3/lib/liblua.so(lua_gc+0x94) [0x7f27efc99c34]
[bt]: (10) /nix/store/5ms06w05g7l9y2rakkf5x23p177qhz67-craftos-pc-2.8.3/lib/liblua.so(+0x23128) [0x7f27efcb1128]
[bt]: (11) /nix/store/5ms06w05g7l9y2rakkf5x23p177qhz67-craftos-pc-2.8.3/lib/liblua.so(+0x10149) [0x7f27efc9e149]
[bt]: (12) /nix/store/5ms06w05g7l9y2rakkf5x23p177qhz67-craftos-pc-2.8.3/lib/liblua.so(+0x1e633) [0x7f27efcac633]
[bt]: (13) /nix/store/5ms06w05g7l9y2rakkf5x23p177qhz67-craftos-pc-2.8.3/lib/liblua.so(+0x103ec) [0x7f27efc9e3ec]
[bt]: (14) /nix/store/5ms06w05g7l9y2rakkf5x23p177qhz67-craftos-pc-2.8.3/lib/liblua.so(+0xf67c) [0x7f27efc9d67c]
[bt]: (15) /nix/store/5ms06w05g7l9y2rakkf5x23p177qhz67-craftos-pc-2.8.3/lib/liblua.so(lua_resume+0x73) [0x7f27efc9e723]
[bt]: (16) /nix/store/5ms06w05g7l9y2rakkf5x23p177qhz67-craftos-pc-2.8.3/lib/liblua.so(+0x24095) [0x7f27efcb2095]
[bt]: (17) /nix/store/5ms06w05g7l9y2rakkf5x23p177qhz67-craftos-pc-2.8.3/lib/liblua.so(+0x24241) [0x7f27efcb2241]
[bt]: (18) /nix/store/5ms06w05g7l9y2rakkf5x23p177qhz67-craftos-pc-2.8.3/lib/liblua.so(+0x10149) [0x7f27efc9e149]
[bt]: (19) /nix/store/5ms06w05g7l9y2rakkf5x23p177qhz67-craftos-pc-2.8.3/lib/liblua.so(+0x1e633) [0x7f27efcac633]
[bt]: (20) /nix/store/5ms06w05g7l9y2rakkf5x23p177qhz67-craftos-pc-2.8.3/lib/liblua.so(+0x103ec) [0x7f27efc9e3ec]
[bt]: (21) /nix/store/5ms06w05g7l9y2rakkf5x23p177qhz67-craftos-pc-2.8.3/lib/liblua.so(+0xf67c) [0x7f27efc9d67c]
[bt]: (22) /nix/store/5ms06w05g7l9y2rakkf5x23p177qhz67-craftos-pc-2.8.3/lib/liblua.so(lua_resume+0x73) [0x7f27efc9e723]
[bt]: (23) /nix/store/5ms06w05g7l9y2rakkf5x23p177qhz67-craftos-pc-2.8.3/lib/liblua.so(+0x24095) [0x7f27efcb2095]
[bt]: (24) /nix/store/5ms06w05g7l9y2rakkf5x23p177qhz67-craftos-pc-2.8.3/lib/liblua.so(+0x24241) [0x7f27efcb2241]
Segmentation fault (core dumped)
```

**Environment (please complete the following information):**
 - OS: NixOS
 - OS Version: Unstable
 - CraftOS-PC Version: v2.8.3
 - Compiled from source? No

**Comment by MCJack123** [comment]
Your fix from #343 caused this ;)

**Comment by simadude** [comment]
Oopsie daisy :3
